Library/Homebrew/cask/cask_loader.rb

```diff
@@ -494,13 +494,26 @@ class FromInstalledPathLoader < FromPathLoader
                 .returns(T.nilable(T.attached_class))
       }
       def self.try_new(ref, warn: false)
-        return unless ref.is_a?(String)
+        token = if ref.is_a?(String)
+          ref
+        elsif ref.is_a?(Pathname)
+          ref.basename(ref.extname).to_s
+        end
+        return unless token
 
-        possible_installed_cask = Cask.new(ref)
+        possible_installed_cask = Cask.new(token)
         return unless (installed_caskfile = possible_installed_cask.installed_caskfile)
 
         new(installed_caskfile)
       end
+
+      sig { params(path: T.any(Pathname, String), token: String).void }
+      def initialize(path, token: T.unsafe(nil))
```

Comment:
```suggestion
      sig { params(path: T.any(Pathname, String), token: String).void }
      def initialize(path, token: "")
```

Reply:
Seems like we aren't even using the `token` parameter in `FromPathLoader`.

Reply:
Yeah looks like it. I'll leave that for another time though because I think that might need a deprecation?